backend/auth/Auth.ts

```diff
@@ -48,8 +75,28 @@ router.post("/login", passport.authenticate("local"), (req, res, next) => {
 
 router.get("/login", renderWebApp);
 
+router.get(
+  "/auth/google",
+  passport.authenticate("google", {
+    scope: ["https://www.googleapis.com/auth/plus.login", "email"],
+  })
+);
+
+router.get(
+  "/auth/google/callback",
+  passport.authenticate("google", { failureRedirect: "/login" }),
+  function (req, res) {
+    res.redirect("/");
+  }
+);
+
 router.use("/", async (req, res, next) => {
-  if (req.session && req.session.passport && req.session.passport.user.id) {
+  if (
```

Comment:
This could be simplified to the following:

```js
if (req?.session?.passport?.user?.id) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining